pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/BaseFilterOperator.java

```diff
@@ -56,4 +73,8 @@ public final BaseFilterBlock getNextBlock(BlockId blockId) {
   }
 
   public abstract BaseFilterBlock nextFilterBlock(BlockId blockId);
+
+  public boolean alwaysFalse() {
```

Comment:
There seems to be 13 implementations of BaseFilterOperator (including tests), I chose to have the default implementation in this class just to avoid code duplication. The new EmptyFilterOperator class I added already overrides this method. Once we have a large number of custom implementations, we can make it abstract. What do you think?